code/game/objects/items/tools/shovel_tools.dm

```diff
@@ -200,4 +203,16 @@
 	. = ..()
 	if(sharp)
 		to_chat(user, "<span class='notice'> This one has been sharpened and can no longer be folded.</span>")
+/obj/item/tool/shovel/unique_action(mob/user)
+	entrenchingmode = !entrenchingmode
+	if(entrenchingmode)
+		user.visible_message("<span class='notice'>[user] reconfigures [src] to entrenching mode.</span>")
+	else
+		user.visible_message("<span class='notice'>[user] reconfigures [src] to digging mode.</span>")
```

Comment:
use early return

if(entrenchimingmode)
      things to do
      return
things to do in the else